akka-remote/src/main/scala/akka/remote/artery/Handshake.scala

```diff
@@ -282,10 +284,8 @@ private[remote] class InboundHandshake(inboundContext: InboundContext, inControl
             // periodically.
             thenInside
           case None =>
-            implicit val ec = materializer.executionContext
-            first.onComplete { _ =>
-              getAsyncCallback[Done](_ => thenInside).invoke(Done)
-            }
+            val callback = getAsyncCallback[Try[Done]](_ => thenInside)
```

Comment:
Would it make sense to create this only once and pass the thunk as well instead of closing over it? I.e. create a static `getAsyncCallback[() => Unit]` which just runs the thunk?

Reply:
makes sense since it is used many times